signer/tests/integration/complete_deposit.rs

```diff
@@ -366,6 +390,18 @@ async fn complete_deposit_validation_invalid_mint_amount() {
     setup.store_deposit_request(&db).await;
     setup.store_deposit_decisions(&db).await;
 
+    // Store outstanding sweep transaction packages in the database, which
+    // includes the above deposit request. But remember that this represents a
+    // sweep transaction that has been broadcast to the mempool, but not yet
+    // observed in a block (we would need to also call `.store_sweep_tx()` for
+    // that).
+    //
+    // Note: we need to store the withdrawal request to satisfy FK's since
+    // `TestSweepSetup` includes a withdrawal in the sweep transaction by
+    // default, but we don't use it.
+    setup.store_withdrawal_request(&db).await;
```

Comment:
We shouldn't need the `setup.store_withdrawal_request(&db).await;` calls in these tests.

Reply:
We shouldn't, but unfortunately we do: when we `store_sweep_transactions`, we write into `sweep_transactions` as well as `swept_deposits` and `swept_withdrawals` for each request swept, and those two tables have FK to `deposit_requests` and `withdrawal_requests`; since `TestSweepSetup` generates both a deposit and a withdrawal, if we don't store the withdrawal request when `store_sweep_transactions` we get
```
insert or update on table "swept_withdrawals" violates foreign key constraint "swept_withdrawals_withdrawal_request_id_withdrawal_request_fkey"
[...]
Key (withdrawal_request_id, withdrawal_request_block_hash)=(0, x86443fba4481842a9a295ad6caea944116087d37305da2adbb9bd6dc48920574) is not present in table "withdrawal_requests"
```

Reply:
Oh whoops, I forgot about that. Okay cool.